stwo_cairo_prover/crates/cairo-air/src/utils.rs

```diff
@@ -28,29 +31,52 @@ pub fn serialize_proof_to_file<MC: MerkleChannel>(
 ) -> Result<(), std::io::Error>
 where
     MC::H: Serialize,
-    <MC::H as MerkleHasher>::Hash: CairoSerialize,
+    <MC::H as MerkleHasher>::Hash: CairoSerialize + CompactBinary,
 {
     let span = span!(Level::INFO, "Serialize proof").entered();
 
-    let mut proof_file = File::create(proof_path)?;
+    let mut proof_file = File::create(&proof_path)?;
 
     match proof_format {
         ProofFormat::Json => {
-            proof_file.write_all(sonic_rs::to_string_pretty(proof)?.as_bytes())?;
+            fs::write(&proof_path, serde_json::to_string(&proof)?)?;
         }
         ProofFormat::CairoSerde => {
             let mut serialized: Vec<starknet_ff::FieldElement> = Vec::new();
             CairoSerialize::serialize(proof, &mut serialized);
-
-            let hex_strings: Vec<String> = serialized
-                .into_iter()
-                .map(|felt| format!("0x{:x}", felt))
-                .collect();
-
-            proof_file.write_all(sonic_rs::to_string_pretty(&hex_strings)?.as_bytes())?;
+            fs::write(&proof_path, serde_json::to_string(&serialized)?)?;
```

Comment:
not sure if i like it or not. The previous way wasn't really good but at least we were sure it'd be hex

Reply:
remove it